include/sample_rate_converter.h

```diff
@@ -0,0 +1,124 @@
+#define SAMPLE_RATE_CONVERT_INPUT_BUF_SIZE 2 * sizeof(uint16_t)
+/* The output buffer must be big enough to store 6 spillover samples */
+#define SAMPLE_RATE_CONVERTER_RINGBUF_SIZE                                                         \
+	(CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE + 6) * sizeof(uint16_t)
+#elif CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_32
+/* The input buffer must be big enough to store 2 spillover samples */
+#define SAMPLE_RATE_CONVERT_INPUT_BUF_SIZE 2 * sizeof(uint32_t)
+/* The output buffer must be big enough to store 6 spillover samples */
+#define SAMPLE_RATE_CONVERTER_RINGBUF_SIZE                                                         \
+	(CONFIG_SAMPLE_RATE_CONVERTER_BLOCK_SIZE + 6) * sizeof(uint32_t)
+#endif
+
+/** Buffer used for storing input bytes to the sample rate converter*/
+struct buf_ctx {
+	uint8_t buf[SAMPLE_RATE_CONVERT_INPUT_BUF_SIZE];
+	size_t bytes_in_buf;
+};
+
+/** Context for the sample rate conversion */
+struct sample_rate_converter_ctx {
+	int input_sample_rate;
+	int output_sample_rate;
+	enum sample_rate_converter_filter filter_type;
+	struct buf_ctx input_buf;
+	struct ring_buf output_ringbuf;
+	uint8_t output_ringbuf_data[SAMPLE_RATE_CONVERTER_RINGBUF_SIZE];
+	union {
+#ifdef CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_16
+		arm_fir_interpolate_instance_q15 fir_interpolate_q15;
+		arm_fir_decimate_instance_q15 fir_decimate_q15;
+#elif CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_32
+		arm_fir_interpolate_instance_q31 fir_interpolate_q31;
+		arm_fir_decimate_instance_q31 fir_decimate_q31;
+#endif
+	};
+#ifdef CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_16
+	q15_t state_buf_15[SAMPLE_RATE_CONVERTER_STATE_BUFFER_SIZE];
+#elif CONFIG_SAMPLE_RATE_CONVERTER_BIT_DEPTH_32
+	q31_t state_buf_31[SAMPLE_RATE_CONVERTER_STATE_BUFFER_SIZE];
+#endif
+};
+
+/**
+ * @brief	Process input samples and produce output samples with new samplerate
+ *
+ * @details	Takes samples with the input sample rate, and converts them to the new requested
+ *		sample rate by filtering the samples before adding or removing samples. The context
+ *		for the sample rate conversion does not need to be initialized before calling
+ *		process, and if any parameters change between calls the context will be
+ *		re-inititalized. As the process has requirements for the number of input samples
+ *		based on the conversion ratio, the module will buffer both input and output bytes
+ *		when needed to meet this criteria.
+ *
+ * @param[in,out]	ctx			Pointer to sample rate conversion context
+ * @param[in]	 	filter 			Filter type to be used for the conversion
+ * @param[in]		input 			Pointer to samples to process
+ * @param[in]		input_size 		Size of the input in bytes
+ * @param[in] 		input_sample_rate 	Sample rate of the input bytes
+ * @param[out]		output 			Array that output will be written
+ * @param[in] 		output_size 		Number of bytes requested as output
+ * @param[in] 		output_sample_rate	Sample rate of output
+ *
```

Comment:
As discussed offline CI has not run on this PR, so this is a bit TBD at the moment. Looking at other header files this looks like a valid format though, so I will keep it until CI says otherwise 😄